src/hub/vendor/customer.py

```diff
@@ -257,159 +256,224 @@ def run(self) -> None:
-                    customer_id=self.payload.data.object.customer,
-                    created=self.payload.data.object.created,
-                    plan_amount=self.payload.data.object.plan.amount,
-                    eventId=self.payload.id,  # required by FxA
-                    canceled_at=self.payload.data.object.canceled_at,
-                    cancel_at=self.payload.data.object.cancel_at,
-                    cancel_at_period_end=self.payload.data.object.cancel_at_period_end,
-                    currency=self.payload.data.object.plan.currency,
-                    current_period_start=self.payload.data.object.current_period_start,
-                    current_period_end=self.payload.data.object.current_period_end,
-                    invoice_number=invoice_number,
-                    brand=brand,
-                    last4=last4,
-                    charge=charge,
-                )
-                logger.info("customer subscription new recurring", data=data)
-                routes = [StaticRoutes.SALESFORCE_ROUTE]
-                self.send_to_routes(routes, json.dumps(data))
-            else:
-                logger.info(
-                    "cancel_at_period_end false",
-                    data=self.payload.data.object.cancel_at_period_end,
-                )
-        else:
-            logger.error(
-                "customer subscription updated - no userid",
-                error=self.payload.object.customer,
-            )
-            raise ClientError(
-                f"userid is None for customer {self.payload.object.customer}"
-            )
+
+        if user_id is None:
+            logger.error("customer subscription updated - no userid", error=customer_id)
+            raise ClientError(f"userid is None for customer {customer_id}")
+
+        return user_id
 
     def get_latest_invoice(self, invoice_id) -> Optional[Dict[str, Any]]:
+        """
+        Fetch the Invoice object from Stripe by ID.
+        If the charge has not been completed, wait and attempt again to fetch the object.
+        If after 2 attempts, the Invoice has still not been completed :return None
+        :param invoice_id:
+        :return:
+        """
         attempts = 2
-        latest_invoice = stripe.Invoice.retrieve(id=invoice_id)
-        logger.info("get latest invoice", latest_invoice=latest_invoice)
         while attempts > 0:
-            attempts -= 1
-            if latest_invoice.charge is None or latest_invoice.number is None:
-                time.sleep(0.2)
-                self.get_latest_invoice(invoice_id)
-            else:
+            latest_invoice = stripe.Invoice.retrieve(id=invoice_id)
+            logger.info("get latest invoice", latest_invoice=latest_invoice)
+
+            if latest_invoice.charge is not None and latest_invoice.number is not None:
                 return latest_invoice
+
+            attempts -= 1
+            time.sleep(0.2)
```

Comment:
In get_latest_invoice() you are implementing exponential backoff, you should use shared/vendor - retrieve_stripe_invoice.

Reply:
removed method and updated references to use shared/vendor method